api/v1alpha1/healthcheck_types.go

```diff
@@ -73,6 +82,12 @@ type HealthCheckList struct {
 	Items           []HealthCheck `json:"items"`
 }
 
+// Workflow struct describes an Argo workflow
+type RemedyWorkflow struct {
+	GenerateName string          `json:"generateName,omitempty"`
```

Comment:
Is the `remedyWorkflow` required to always have a `generateName`? What if someone provides just the `name`?

Reply:
if we pass only name then we will get an error from argo workflow:
2020-07-16T12:59:11.368Z	ERROR	controllers.HealthCheck	Error creating remedyworkflow	{"HealthCheck": "health/fail-healthcheck-cfb54", "error": "Workflow.argoproj.io \"\" is invalid: metadata.name: Required value: name or generateName is required"}

Reply:
`generateName` is used by argo workflow as a prefix for the workflow name and pod name